common/dal/src/accessors/aws_s3/s3.rs

```diff
@@ -53,15 +53,31 @@ impl S3 {
 
         let client = match Self::credential_provider(access_key_id, secret_accesses_key) {
             Some(provider) => Client::new_with(provider, dispatcher),
-            None => Client::new_with(
-                DefaultCredentialsProvider::new().map_err(|e| {
-                    ErrorCode::DALTransportError(format!(
-                        "failed to create default credentials provider, {}",
-                        e
-                    ))
-                })?,
-                dispatcher,
-            ),
+            None => {
+                // check on k8s admission webhook injection
+                if std::env::var("AWS_WEB_IDENTITY_TOKEN_FILE").is_ok() {
```

Comment:
Why not add a new s3 storage config?

Reply:
does not seems to be make sense... using SNS web identity means to use token generated by some controllers with fixed env variable in every instance, there seems have little space on self customization,

Reply:
If we inline a env in the codes, it looks very tricky, and feels like a back door was left open, we should make this configuration public although we are using it from env

Reply:
https://www.kubeflow.org/docs/distributions/aws/deploy/install-kubeflow/#option-1-use-aws-iam-roles-for-service-accounts-default-and-recommended

I would like to add similar feature flag to let user use IAM policies.

Reply:
Nice, thank you @ZhiHanZ